ntrans.py

```diff
@@ -1,107 +1,75 @@
-import nltk
 import translatepy
-import re
-import collections
 import csv
-# import ssl
 
+"""
+DEVELOPER INFO:
 
-def prepare_corpus_data(data_size=1000000):
-    # TODO: Put code that downloads BNC here
-
-    bnc_corpus = nltk.corpus.BNCCorpusReader(root='BNC/Texts/', fileids=r'[A-K]/\w*/\w*\.xml')
-
-    bnc_sents = []
-
-    # TODO:
-    # If user wants a smaller data set than the full corpus, it would be better
-    # to pick random sents() from bnc_corpus instead of for looping over the first
-    # X sents. If data_size = 10.000, pick 10.000 random sents from bnc_corpus
-
-    for count, sentence in enumerate(bnc_corpus.sents()):
-        string_sent = ' '.join(sentence)
-        bnc_sents.append(string_sent)
-        if count == 1000000:  # TODO: Change count == 10000 to "count == data_size"
-            print(count)
-            break
-        # TODO: Remake print(count) to progress bar in tkinter
-        if count % 1000 == 0:
-            print(count)
+The usage of this script relies on the N-gram files in "./ngrams"
 
-    return ' '.join(bnc_sents)
+These files are created through ntrans_dataprep.py
+"""
 
 
 def create_csv_file(source_target_pairs, save_path=None):
+    """
+    Writes source/target pairs to a csv-file
+    """
+
     path = "/Users/Writing/Desktop/"  # TODO: Make path depend on user input and use "save_path"
     filename = "ntrans-glossary.csv"  # TODO: Make filename depend on user input
     full_path = path + filename
 
-    with open(full_path, mode='w') as write_ntrans_file:
-        data_writer = csv.writer(write_ntrans_file, delimiter=',')
+    with open(full_path, mode="w") as write_ntrans_file:
+        data_writer = csv.writer(write_ntrans_file, delimiter=",")
 
-        data_writer.writerow(('English', 'Swedish'))
+        data_writer.writerow(("English", "Swedish"))
         for source_target_pair in source_target_pairs:
             data_writer.writerow(source_target_pair)
 
     return print("N-Trans CSV Glossary has been successfully saved to " + full_path)
 
 
-def machine_translate_ngrams(list_of_ngrams, language=None):
+def machine_translate_ngrams(ngrams):
+    """
+    Translates each N-gram and appends the source/target pair to a list.
+    """
+    source_target_pairs = []
 
     translator = translatepy.Translator()
 
-    source_target_pairs = []
-
-    for enum_count, source_string in enumerate(list_of_ngrams, start=1):
-        print(f"Translating item {enum_count}/{len(list_of_ngrams)}")
-        translated_string = translator.translate(
-            source_string, "Swedish"
-        )  # TODO: Change to language variable
-        # print(source_string + " --> " + str(translated_string))
-        source_target_pairs.append((source_string, str(translated_string).lower()))
+    for n in ngrams.keys():
+        for enum, source_ngram in enumerate(ngrams[n], start=1):
```

Comment:
Nit: Consider using `ngrams.items()`, so that you get the key and the value, instead of looking up the value as `ngrams[n]`.